projects/core/src/button/button.element.ts

```diff
@@ -92,25 +107,26 @@ export class CdsButton extends CdsBaseButton {
   }
 
   update(props: PropertyValues<this>) {
-    if (props.has('loadingState')) {
-      if (this.isDefaultLoadingState(props.get('loadingState') as string)) {
-        // track prior disabled state to set prior value after button is re-enabled from a loading state
-        this._disabled = this.disabled;
-      }
-
-      if (props.get('loadingState') !== undefined) {
-        if (this.isDefaultLoadingState(this.loadingState)) {
-          this.enableButton();
-        } else {
-          this.disableButton();
-        }
+    if (props.has('loadingState') && props.get('loadingState') !== undefined) {
+      if (this.isDefaultLoadingState(this.loadingState)) {
+        this.restoreButton();
+      } else {
+        this.disableButton();
       }
     }
 
     super.update(props);
   }
 
-  private _disabled = false;
+  private disableButton() {
+    this.style.width = getElementWidth(this);
+    super.disabled = true;
+  }
+
+  private restoreButton() {
+    this.style.removeProperty('width');
+    super.disabled = this._disabled_externally;
+  }
```

Comment:
Can we put these private methods back where they were in the file? And rename `restoreButton` back to `enableButton`?

Reply:
Not only were these methods moved up in the file for some reason, they are now before a public method.

Reply:
> And rename `restoreButton` back to `enableButton`?

Nevermind, I see why you renamed it now.